crates/block-producer/src/txqueue/mod.rs

```diff
@@ -94,7 +94,7 @@ where
                 return;
             }
 
-            locked_ready_queue.drain(..).collect()
+            locked_ready_queue.drain(..).rev().collect()
```

Comment:
Why do we need this `.rev()` here?

Reply:
@bobbinth our previous solution processed transactions in reverse order (LIFO) and shuffled them in a case of error during batch building. I changed this ordering to FIFO with keeping the same order in a case of error. Processing transaction in FIFO order is important in order to reduce batch building errors with unathorized notes. 

For example:
1. Let transaction `B` consumes unauthorized note from transaction `A`. 
2. If transaction `B` is processed first and the distance between these transaction in the queue is high enough for them to be put into separate batches, transaction `A` won't be in the ready batches list at the time of processing transaction `B`.
3. This will lead to batch building error and unprocessed transaction will be inserted to the end of the queue.
4. Then transaction `A` will be processed and added to the ready batches list.
5. Eventually transaction `B` will be also processed.

My current solution relies on transaction ordering in order to eliminate such errors. My only concern, is it possible  for transaction `B` to be put into the queue before transaction `A`? In this case the whole queue processing will stop because of constant "unauthorized notes not found" errors. If such is possible, I'd propose to do shuffling of the queue as we did before my changes.